bundles/org.openhab.binding.airgradient/src/main/java/org/openhab/binding/airgradient/internal/handler/AirGradientAPIHandler.java

```diff
@@ -0,0 +1,401 @@
+        pollingJob = scheduler.scheduleWithFixedDelay(this::pollingCode, 5, config.refreshInterval, TimeUnit.SECONDS);
+    }
+
+    private static String getMeasureId(Measure measure) {
+        String id = measure.getLocationId();
+        if (id.isEmpty()) {
+            // Local devices don't have location ID.
+            id = measure.getSerialNo();
+        }
+
+        return id;
+    }
+
+    protected void pollingCode() {
+        List<Measure> measures = getMeasures();
+        Map<String, Measure> measureMap = measures.stream().collect(Collectors.toMap((m) -> getMeasureId(m), (m) -> m));
+
+        for (Thing t : getThing().getThings()) {
+            AirGradientLocationHandler handler = (AirGradientLocationHandler) t.getHandler();
+            if (handler != null) {
+                String locationId = handler.getLocationId();
+                @Nullable
+                Measure measure = measureMap.get(locationId);
+                if (measure != null) {
+                    handler.setMeasurment(locationId, measure);
+                } else {
+                    logger.debug("Could not find measures for location {}", locationId);
+                }
+            }
+        }
+    }
+
+    /**
+     * Return location ids we already have things for.
+     * 
+     * @return location ids we already have things for.
+     */
+    public List<String> getRegisteredLocationIds() {
+        List<Thing> things = getThing().getThings();
+        List<String> results = new ArrayList<>(things.size());
+        for (Thing t : things) {
+            AirGradientLocationHandler handler = (AirGradientLocationHandler) t.getHandler();
+            if (handler != null) {
+                results.add(handler.getLocationId());
+            }
+        }
+
+        return results;
+    }
+
+    /**
+     * Return list of measures from AirGradient API.
+     *
+     * @return list of measures
+     */
+    public List<Measure> getMeasures() {
+        try {
+            ContentResponse response = httpClient.GET(generateMeasuresUrl());
+            String contentType = response.getMediaType();
+            logger.debug("Got measurements with status {}: {} ({})", response.getStatus(),
+                    response.getContentAsString(), contentType);
+            if (response.getStatus() == 200) {
+                updateStatus(ThingStatus.ONLINE);
+                String stringResponse = response.getContentAsString().trim();
+
+                if (CONTENTTYPE_JSON.equals(contentType)) {
+                    return parseJson(stringResponse);
+                } else if (CONTENTTYPE_TEXT.equals(contentType)) {
+                    return parsePrometheus(stringResponse);
+                } else if (CONTENTTYPE_OPENMETRICS.equals(contentType)) {
+                    return parsePrometheus(stringResponse);
+                }
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, response.getContentAsString());
```

Comment:
In general i would not just route the response directly to the thing state, As this may contain large tekst, html tags, json etc.  Better to create a generic http status message + status number. 
Debug information can be added, (headers and so on) or trace (request/response content)

Reply:
Thanks, added status code